impeller/compiler/compiler.cc

```diff
@@ -115,15 +115,13 @@ static CompilerBackend CreateMSLCompiler(
 static CompilerBackend CreateVulkanCompiler(
     const spirv_cross::ParsedIR& ir,
     const SourceOptions& source_options) {
-  // TODO(dnfield): It seems like what we'd want is a CompilerGLSL with
-  // vulkan_semantics set to true, but that has regressed some things on GLES
-  // somehow. In the mean time, go back to using CompilerMSL, but set the Metal
-  // Language version to something really high so that we don't get weird
-  // complaints about using Metal features while trying to build Vulkan shaders.
-  // https://github.com/flutter/flutter/issues/123795
-  return CreateMSLCompiler(
-      ir, source_options,
-      spirv_cross::CompilerMSL::Options::make_msl_version(3, 0, 0));
+  auto vk_compiler = std::make_shared<spirv_cross::CompilerGLSL>(ir);
```

Comment:
Do we know how this change had previously broken GLES and how this changed fixed it?

Reply:
The original issue was light on details but there is now a separate `CreateGLSLCompiler` which is probably what's used. So this change shouldn't affect GLSL at all. I believe earlier there used to be a common compiler instance creation and I can believe was hard to configure just right for one backend without causing unintended consequences elsewhere. But again, the issue was light on details so its hard to say. I'll keep an eye out for repercussions.